src/mlpack/methods/ann/layer/c_relu_impl.hpp

```diff
@@ -0,0 +1,95 @@
+/**
+ * @file methods/ann/layer/c_relu_impl.hpp
+ * @author Jeffin Sam
+ *
+ * Implementation of CReLU layer.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_C_RELU_IMPL_HPP
+#define MLPACK_METHODS_ANN_LAYER_C_RELU_IMPL_HPP
+
+// In case it hasn't yet been included.
+#include "c_relu.hpp"
+
+namespace mlpack {
+
+template<typename MatType>
+CReLUType<MatType>::CReLUType() :
+    Layer<MatType>()
+{
+  // Nothing to do here.
+}
+
+template<typename MatType>
+CReLUType<MatType>::CReLUType(
+    const CReLUType& other) :
+    Layer<MatType>(other)
+{
+  // Nothing to do here.
+}
+
+template<typename MatType>
+CReLUType<MatType>::CReLUType(
+    CReLUType&& other) :
+    Layer<MatType>(std::move(other))
+{
+  // Nothing to do here.
+}
+
+template<typename MatType>
+CReLUType<MatType>&
+CReLUType<MatType>::operator=(const CReLUType& other)
+{
+  if (&other != this)
+  {
+    Layer<MatType>::operator=(other);
+  }
+
+  return *this;
+}
+
+template<typename MatType>
+CReLUType<MatType>&
+CReLUType<MatType>::operator=(CReLUType&& other)
+{
+  if (&other != this)
+  {
+    Layer<MatType>::operator=(std::move(other));
+  }
+
+  return *this;
+}
+
+template<typename MatType>
+void CReLUType<MatType>::Forward(
+    const MatType& input, MatType& output)
+{
+  output = arma::join_cols(arma::max(input, 0.0 * input), arma::max(
+      (-1 * input), 0.0 * input));
```

Comment:
This is a bit of a strange implementation, and I don't think it will be efficient.  (I know that it is the original implementation; it's just something I saw while passing through the review.)  Here's one I think would work better (although it is more verbose!).  I borrowed from `LeakyReLUType`.
```suggestion
  #pragma omp for
  for (size_t i = 0; i < (size_t) input.n_cols; ++i)
  {
    for (size_t j = 0; j < (size_t) input.n_rows; ++j)
    {
      output(j, i) = std::max(input(j, i), 0.0);
      output(j + input.n_rows, i) = std::max(-input(j, i), 0.0);
    }
  }
```